.github/workflows/envoy-macos.yml

```
@@ -48,7 +48,7 @@ jobs:
       command:
       container-command:
       request: ${{ needs.load.outputs.request }}
-      runs-on: macos-12-xl
+      runs-on: macos-14-xl
```

```suggestion
      runs-on: macos-14-xlarge
```

trying to make sense of docs - which are at least incomplete (eg `macos-12-xl` is mentioned elsewhere and works) here:

https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/running-jobs-on-larger-runners?platform=mac#available-macos-larger-runners

one thing im noticing is that `large` <> `xlarge` seem to have their names in reverse (large is "larger" than x-large!!!) 

also seems to relate more to cpu type

anyhow testing elsewhere and `macos-14-xlarge` works insofar as it runs - ci didnt fail yet - but taking some time

---

ci passed - @ggreenway ill post you the link offline